### PR TITLE
`flutter/lib/src/`: refactoring if-chains into switch expressions

### PR DESCRIPTION
**Pull request by nate-thegrate**
Based on issue #144903, this pull request aims to bring the codebase more in line with the [Flutter repo style guide](https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#avoid-using-if-chains-or--or--with-enum-values):

> ### Avoid using `if` chains or `?:` or `==` with enum values

### CODE REVIEW

**Review by goderbauer — APPROVED** [review]
LGTM

**Review by Piinks — APPROVED** [review]
Flutter_LGTM
Thank you!

File: packages/flutter/lib/src/material/input_decorator.dart [hunk 1]
```diff
@@ -562,13 +562,11 @@ class FloatingLabelAlignment {
...
   static String _stringify(double x) {
-    if (x == -1.0) {
-      return 'FloatingLabelAlignment.start';
-    }
-    if (x == 0.0) {
-      return 'FloatingLabelAlignment.center';
-    }
-    return 'FloatingLabelAlignment(x: ${x.toStringAsFixed(1)})';
+    return switch (x) {
+     -1.0 => 'FloatingLabelAlignment.start',
```
Comment (QuncCccccc) [inline packages/flutter/lib/src/material/input_decorator.dart:566]: nit: should have 2-character indentation:)
```suggestion
      -1.0 => 'FloatingLabelAlignment.start',
```

**Review by QuncCccccc — APPROVED** [review]
LGTM! Thanks a lot for your contribution:)!

File: packages/flutter/lib/src/material/input_decorator.dart (hunk 1 above)
Comment (nate-thegrate) [reply packages/flutter/lib/src/material/input_decorator.dart:566]: It looks like auto-submit bot kicked in right when you submitted the review 😒

I thought it looked nice with all the numbers aligned, but you're right—in the future I'll be better about following this convention!

Comment (QuncCccccc) [reply packages/flutter/lib/src/material/input_decorator.dart:566]: Ah no worries at all! Thanks:)!

**Pull request merged**